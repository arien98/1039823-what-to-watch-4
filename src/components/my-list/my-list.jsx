import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {getFavoriteFilms} from '../../reducer/data/selectors.js';
import {Operation as DataOperation} from '../../reducer/data/data.js';
import Header from '../header/header.jsx';
import Footer from '../footer/footer.jsx';
import FilmsList from '../films-list/films-list.jsx';
import ErrorScreen from '../error-screen/error-screen.jsx';

class MyList extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    const {loadFavoriteFilms} = this.props;
    loadFavoriteFilms();
  }

  render() {
    const {favoriteFilms, onFilmCardClick} = this.props;

    // if (!favoriteFilms) {
    //   return <ErrorScreen errorText={`Loading...`}/>;
    // }

    return (
      <div className="user-page">

        <Header>
          <h1 className="page-title user-page__title">My list</h1>
        </Header>

        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <FilmsList
            filmsData={favoriteFilms}
            onFilmCardClick={onFilmCardClick}
          />
        </section>

        <Footer />
      </div>
    );
  }
}

MyList.propTypes = {
  favoriteFilms: PropTypes.array,
  onFilmCardClick: PropTypes.func.isRequired,
  loadFavoriteFilms: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  favoriteFilms: getFavoriteFilms(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadFavoriteFilms() {
    dispatch(DataOperation.loadFavoriteFilms());
  },
});


export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);
