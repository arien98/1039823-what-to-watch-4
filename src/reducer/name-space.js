export const NameSpace = {
  DATA: `DATA`,
  SCREEN: `SCREEN`,
  USER: `USER`,
};
