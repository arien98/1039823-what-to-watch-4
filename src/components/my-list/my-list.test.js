import React from "react";
import renderer from "react-test-renderer";
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {history} from '../../history.js';
import {NameSpace} from "../../reducer/name-space.js";
import {MyList} from "./my-list.jsx";

const mockStore = configureStore([]);

const props = {
  loadFavoriteFilms: () => {},
  onFilmCardClick: () => {},
  favoriteFilms: [
    {id: 1, title: `Mindhunter`, smallPoster: `http://placekitten.com/245/175`, poster: `http://placekitten.com/245/175`, bigPoster: `img/bg-the-grand-budapest-hotel.jpg`, genre: `drama`, releaseDate: 2000, video: ``, preview: ``, ratingScore: 5, ratingNumber: 100, ratingValue: ``, starring: [``, `1`, `3`], director: ``, reviews: []},
    {id: 2, title: `Aviator`, smallPoster: `http://placekitten.com/245/175`, poster: `http://placekitten.com/245/175`, bigPoster: `img/bg-the-grand-budapest-hotel.jpg`, genre: `drama`, releaseDate: 2000, video: ``, preview: ``, ratingScore: 5, ratingNumber: 100, ratingValue: ``, starring: [``, `1`, `3`], director: ``, reviews: []},
    {id: 3, title: `Pulp Fiction`, smallPoster: `http://placekitten.com/245/175`, poster: `http://placekitten.com/245/175`, bigPoster: `img/bg-the-grand-budapest-hotel.jpg`, genre: `drama`, releaseDate: 2000, video: ``, preview: ``, ratingScore: 5, ratingNumber: 100, ratingValue: ``, starring: [``, `1`, `3`], director: ``, reviews: []},
  ],
};

it(`AddReview page should render correctly`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: `NO_AUTH`,
      authorizationInfo: {
        id: 0,
        email: ``,
        name: ``,
        avatar: ``,
      }
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <MyList
              {...props}
            />
          </Router>
        </Provider>, {
          createNodeMock: () => {
            return {};
          }
        })
      .toJSON();

  expect(tree).toMatchSnapshot();
});
