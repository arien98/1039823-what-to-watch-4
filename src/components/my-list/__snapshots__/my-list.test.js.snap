// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReview page should render correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          data-id={1}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Mindhunter"
            className="small-movie-card__pic"
            height="175"
            src="http://placekitten.com/245/175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
          onClick={[Function]}
        >
          <a
            className="small-movie-card__link"
            data-id={1}
            href="movie-page.html"
          >
            Mindhunter
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          data-id={2}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Aviator"
            className="small-movie-card__pic"
            height="175"
            src="http://placekitten.com/245/175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
          onClick={[Function]}
        >
          <a
            className="small-movie-card__link"
            data-id={2}
            href="movie-page.html"
          >
            Aviator
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
      >
        <div
          className="small-movie-card__image"
          data-id={3}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <img
            alt="Pulp Fiction"
            className="small-movie-card__pic"
            height="175"
            src="http://placekitten.com/245/175"
            width="280"
          />
        </div>
        <h3
          className="small-movie-card__title"
          onClick={[Function]}
        >
          <a
            className="small-movie-card__link"
            data-id={3}
            href="movie-page.html"
          >
            Pulp Fiction
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
